From f199ce66fd93e649ee9369b7829116c48117e92b Mon Sep 17 00:00:00 2001
From: Harshesh Valera <harshesh.v@variscite.com>
Date: Wed, 23 Feb 2022 12:46:20 -0800
Subject: [PATCH] camera: Fix 60 fps request for video recording preview

Due to AF avaialble for OV5640 the HAL tries to request 60fps preview

Signed-off-by: Harshesh Valera <harshesh.v@variscite.com>
---
 camera/CameraMetadata.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/camera/CameraMetadata.cpp b/camera/CameraMetadata.cpp
index 45e8592..f29495f 100644
--- a/camera/CameraMetadata.cpp
+++ b/camera/CameraMetadata.cpp
@@ -1054,6 +1054,9 @@ status_t CameraMetadata::createSettingTemplate(std::unique_ptr<HalCameraMetadata
         // Hardware level at least "limited", need fps fixed.
         static const int32_t aeTargetFpsRange[2] = {60, 60};
         base->Set(ANDROID_CONTROL_AE_TARGET_FPS_RANGE, aeTargetFpsRange, 2);
+    } else {
+        static const int32_t aeTargetFpsRange[2] = {30, 30};
+        base->Set(ANDROID_CONTROL_AE_TARGET_FPS_RANGE, aeTargetFpsRange, 2);
     }
 
     return OK;
-- 
2.25.1

